napari/layers/base/base.py

```diff
@@ -518,30 +518,37 @@ def blending(self, blending):
         self.events.blending()
 
     @property
-    def visible(self):
+    def visible(self) -> bool:
         """bool: Whether the visual is currently being displayed."""
         return self._visible
 
     @visible.setter
-    def visible(self, visibility):
-        self._visible = visibility
+    def visible(self, visible: bool):
+        self._visible = visible
         self.refresh()
         self.events.visible()
-        self.editable = self._set_editable() if self.visible else False
 
     @property
-    def editable(self):
+    def editable(self) -> bool:
         """bool: Whether the current layer data is editable from the viewer."""
         return self._editable
 
     @editable.setter
-    def editable(self, editable):
+    def editable(self, editable: bool):
         if self._editable == editable:
             return
         self._editable = editable
-        self._set_editable(editable=editable)
+        self._on_editable_changed()
         self.events.editable()
 
+    def _reset_editable(self) -> None:
```

Comment:
I separated the potential mutation that's dependent on layer state, and the side-effect of an actual change in the value of `editable` to help clarify the intent here. Let me know if I misunderstood.